packages/form-core/src/FormApi.ts

```diff
@@ -37,13 +37,19 @@ export type FieldInfo<TFormData> = {
 export type ValidationMeta = {
   validationCount?: number
   validationAsyncCount?: number
-  validationPromise?: Promise<ValidationError>
-  validationResolve?: (error: ValidationError) => void
-  validationReject?: (error: unknown) => void
+  validationPromise?: Promise<ValidationError[]>
+  validationResolve?: (errors: ValidationError[]) => void
+  validationReject?: (errors: unknown) => void
 }
 
 export type ValidationError = undefined | false | null | string
 
+export type ValidationErrorMapKeys = 'onChange' | 'onSubmit' | 'onBlur'
```

Comment:
I'm in favor of changing this to:

```ts
export type ValidationErrorMapKeys = `on${Capitalize<ValidationCause>}`
```

But I could be convinced either way. What do you think?

Reply:
Oh interesting, I didn't about this in TS. Learned something new today! Thanks, I prefer your suggestion actually, will update!